src/main/scala/faces/utils/Helpers.scala

```diff
@@ -287,15 +290,398 @@ case class Helpers(cfg: FacesSettings)(implicit rnd: Random) {
     withLight
   }
 
-  def writeImg(img: PixelImage[RGBA], id: Int, n: Int, postfix: String): Unit = {
+  lazy val loadTextures: IndexedSeq[File] = {
+    // search all texture files
+    new File(inTexturesPath).listFiles.filter(_.getName.endsWith(".png")).toIndexedSeq
+  }
+
+  def writeImg(img: PixelImage[RGBA], id: Int, n: Int, postfix: String, mask: PixelImage[RGBA]): PixelImage[RGBA] = {
     val outImgPathID = outImgPath + id + "/"
     if (!Path(outImgPathID).exists) {
       Path(outImgPathID).createDirectory(failIfExists = false)
     }
     PixelImageIO.write(img.map { f => f.toRGB }, new File(outImgPathID + id + "_" + n + postfix + ".png"))
+    var occlusion_mask = mask
+
+    if(cfg.occlusion.renderOcclusion==true && postfix.equals("")) {
+      val outOccPathID = outOccPath + id + "/"
+      if (!Path(outOccPathID).exists) {
+        Path(outOccPathID).createDirectory(failIfExists = false)
+      }
+
+      val outMaskPathID = outMaskPath + id + "/"
+      if (!Path(outMaskPathID).exists) {
+        Path(outMaskPathID).createDirectory(failIfExists = false)
+      }
+
+      // New 'occlusionMode'
```

Comment:
please add meaningfull comments instead of this placeholder (also for following)